frontend/src/components/UI/Tooltip.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import SamwiseIcon from './SamwiseIcon';
+import { IconName } from './Samwise-icon-types';
+
+type props = {
+  readonly name: string;
+  readonly iconName: string;
+}
+type states = {
+  name: string;
+  iconName: IconName;
+  hover: boolean;
+}
+
+class Tooltip extends React.Component<props, states> {
+  public constructor(props: any) {
+    super(props);
+    this.state = {
+      name: props.name,
+      iconName: props.iconName,
```

Comment:
You are duplicating props' field into state, which can easily go out of sync. Do you have compelling reasons to do that?